Fixed constant names and changed ResultCallable to Object

raddmag opened the pull request:
Changed result callable to object in order to allow a setter for it in
Uploader.

Also, I thought that params was just going to be an array of JSON
objects, this allows you to store different types that aren't
necessarily string types. What is the benefit of the other way?

raddmag committed:
Fixed constant names and changed ResultCallable to Object

Changed result callable to object in order to allow a setter for it in
Uploader.

Also, I thought that params was just going to be an array of JSON
objects, this allows you to store different types that aren't
necessarily string types. What is the benefit of the other way?

kmh287 commented:
My point is that the uploader should be agnostic to the inner structure of the JSONObjects. So if a JSONObject has nested JSONObjects, it should still work.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/ResultCallable.java:
```diff
@@ -1,10 +1,8 @@
 package cs5150athletetracking.com.athletetracking;
```
This should remain an interface.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,107 @@
+        /* Sent LocationJson. Check for received confirmation */
+
+        if(result == SUCCESS) {
+            getCallBack().success();
```
Need to null check the callback

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,107 @@
+            getCallBack().success();
+        } else{
+            Log.e(TAG, "Failed to confirm communication with server.");
+            getCallBack().failure();
```
null check

raddmag committed:
Changed parameters back to LocationJSON

Added HashTable to LocationJSON.
This allows you construct LocationJSON’s as before, and lets me treat
it as a JSONObject.

Taking in a complete abstract JSON structure would not be ideal. It’s
better to define the types of JSON’s we’ll be working with.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    @Override
+    protected Integer doInBackground(LocationJSON... params) {
```
Needs to be JSONObject. See my other comment on this

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,109 @@
+
+                // Format LocationJSON argument for the POST method
+                // The format is: "key1=value1&key2=value2"
+                while (keys.hasNext()) {
```
Rather than do all of this, which only looks like it will open the first level of the JSON, why not just get the toString() representation of the entire JSON and upload that (by writing to the output stream). Any half-good JSON library on the backend will be able to parse a JSON with arbitrary structure from a string representation.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/LocationJSON.java:
```diff
@@ -33,13 +37,20 @@ public String toString(){
+                        double latitude, double longitude,
                         double altitude){
         try {
+            this.params = new LinkedHashMap<>();
```
No point in this being a JSON if we're just going to use a map and use the current uploader strategy. I'm sure we can just upload the JSON string directly without all of this extra work that ultimately limits our JSON structure.

kmh287 commented:
http://stackoverflow.com/questions/7181534/http-post-using-json-in-java

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/LocationJSON.java:
```diff
@@ -33,13 +37,20 @@ public String toString(){
             this.put(SERIALIZE_KEY.TIMESTAMP.toString(),timestamp);
             this.put(SERIALIZE_KEY.LATITUDE.toString(), latitude);
-            this.put(SERIALIZE_KEY.LONGITUDE.toString(),longitutde);
+            this.put(SERIALIZE_KEY.LONGITUDE.toString(),longitude);
```
good catch

raddmag committed:
Uploads JSON string directly

kmh287 commented:
Hey everything looks good. Are you able to test this out and see if it works? (I'm just curious if we can get some endpoint to receive some arbitrary JSON)
